pro/src/new_components/OfferIndividualForm/hooks/useFilteredVenueList/__specs__/useFilteredVenueList.spec.ts

```diff
@@ -0,0 +1,111 @@
+    id: 'CC',
+    name: 'Lieu online CC',
+    managingOffererId: 'A',
+    isVirtual: true,
+  }
+
+  beforeEach(() => {
+    const venueList = [
+      {
+        id: 'AA',
+        name: 'Lieu offline AA',
+        managingOffererId: 'A',
+        isVirtual: false,
+      },
+      {
+        id: 'BB',
+        name: 'Lieu offline BB',
+        managingOffererId: 'A',
+        isVirtual: false,
+      },
+    ]
+    const subCategories = [
+      {
+        id: 'A-A',
+        categoryId: 'A',
+        proLabel: 'Sous catégorie online de A',
+        isEvent: false,
+        conditionalFields: [],
+        canBeDuo: false,
+        canBeEducational: false,
+        onlineOfflinePlatform: CATEGORY_STATUS.ONLINE,
+        reimbursementRule: REIMBURSEMENT_RULES.STANDARD,
+        isSelectable: true,
+      },
+      {
+        id: 'A-B',
+        categoryId: 'A',
+        proLabel: 'Sous catégorie offline de A',
+        isEvent: false,
+        conditionalFields: [],
+        canBeDuo: false,
+        canBeEducational: false,
+        onlineOfflinePlatform: CATEGORY_STATUS.OFFLINE,
+        reimbursementRule: REIMBURSEMENT_RULES.STANDARD,
+        isSelectable: true,
+      },
+      {
+        id: 'A-C',
+        categoryId: 'A',
+        proLabel: 'Sous catégorie online or offline de A',
+        isEvent: false,
+        conditionalFields: [],
+        canBeDuo: false,
+        canBeEducational: false,
+        onlineOfflinePlatform: CATEGORY_STATUS.ONLINE_OR_OFFLINE,
+        reimbursementRule: REIMBURSEMENT_RULES.STANDARD,
+        isSelectable: true,
+      },
+    ]
+
+    props = {
+      subCategories,
+      subcategoryId: '',
+      venueList,
+    }
+  })
+
+  it('should not filter venue on isVirtual when subCatagory is not selected', async () => {
+    props.venueList.push(virtualVenue)
+    const { result } = renderHook(() => useFilteredVenueList(props))
+    expect(result.current.length).toEqual(3)
+    expect(result.current[0].id).toEqual('AA')
+    expect(result.current[1].id).toEqual('BB')
+    expect(result.current[2].id).toEqual('CC')
+  })
+  it('should not filter venue on isVirtual when subCatagory is ONLINE or OFFLINE', async () => {
+    props.subcategoryId = 'A-C'
+    props.venueList.push(virtualVenue)
+    const { result } = renderHook(() => useFilteredVenueList(props))
+    expect(result.current.length).toEqual(3)
+    expect(result.current[0].id).toEqual('AA')
+    expect(result.current[1].id).toEqual('BB')
+    expect(result.current[2].id).toEqual('CC')
+  })
+  it('should filter venue on isVirtual when subCatagory is ONLINE only', async () => {
+    props.subcategoryId = 'A-A'
+    props.venueList.push(virtualVenue)
+    const { result } = renderHook(() => useFilteredVenueList(props))
+    expect(result.current.length).toEqual(1)
+    expect(result.current[0].id).toEqual('CC')
+  })
+  it('should filter venue on not isVirtual when subCatagory is OFFLINE only', async () => {
+    props.subcategoryId = 'A-B'
```

Comment:
ici aussi, une copie de l'objet permettrait de garder les tests en isolation sans modifier l'objet initial

Reply:
Un nouvelle objet et généré à chaque `it`, les deux notations reviennent au même.
Mais je vais changer si tu préféré la notation que tu propose.

Reply:
ah oui effectivement c'est dans le beforeEach
mais si tu le mets dans beforeAll ça ne fonctionne plus, ça me semble un peu risqué de répandre cette façon de faire dans le repo